synapse/storage/roommember.py

```diff
@@ -45,6 +45,7 @@ class ProfileInfo:
     display_name: Optional[str]
 
 
+# TODO This is used as a cached value and is mutable.
 @attr.s(slots=True, frozen=True, weakref_slot=False, auto_attribs=True)
 class MemberSummary:
```

Comment:
Is it? `frozen=True` here. Oh, but `members` is a List.

Reply:
Yes, it is due to `members`. The code is hard to refactor to make that immutable, unfortunately.